source/ClientRequest/AMDSClientContinuousDataRequest.cpp

```diff
@@ -57,13 +69,30 @@ AMDSClientContinuousDataRequest& AMDSClientContinuousDataRequest::operator =(con
 	return (*this);
 }
 
+void AMDSClientContinuousDataRequest::setSocketKey(const QString &socketKey)
+{
+	AMDSClientDataRequest::setSocketKey(socketKey);
+
+	foreach (QString bufferName, bufferNames()) {
+		AMDSClientContinuousDataRequest * dataRequest = bufferDataRequestList_.value(bufferName);
+		if (dataRequest)
+			dataRequest->setSocketKey(socketKey);
+	}
+}
+
+void AMDSClientContinuousDataRequest::setBufferNames(const QStringList &names) {
+	bufferNameList_.clear();
```

Comment:
Because it is a QStringList, it's smart and will make a deep copy when it needs to.  Syntactically, what I suggested does exactly what yours does but in fewer steps.